### ISSUE TITLE
Erroneous landmark-main-is-top-level and landmark-banner-is-top-level errors reported on valid markup

### ISSUE THREAD

**Issue by bjankord** [dequelabs/axe-core#1583]
**Expectation:** That axe-core does not report an error when a header and main element are used in a webpage and they are not within any landmark containers.

**Actual:** axe is reporting errors about `landmark-main-is-top-level` and `landmark-banner-is-top-level` when I use https://www.deque.com/axe/axe-for-web/playground/ and the following markup:

```html
<html>
  <body>
    <header role="banner">
      <p>This is the header content</p>
    </header>
    <main role="main">
      <p>This is the main content</p>
    </main>
  </body>
</html>
```

**Motivation:** This appears to be valid markup that should not trigger an axe error.

<pre><code>
axe-core version: Whatever version of axe-core is used with https://www.deque.com/axe/axe-for-web/playground/
</code></pre>

I've created a reduced test case on codepen as well: https://codepen.io/bjankord/pen/pmpNpR

Note that the Chrome axe and axe-Coconut extensions do not report this error.

**Comment by WilcoFiers** [comment]
This problem is caused by the way the playground is set up, not the way axe-core works. I'll see that it gets fixed.

**Comment by bjankord** [comment]
I'm curious about the root problem as I originally saw it in some of our webdriver.io tests that inject axe-core into the page to report issues. The only other place I've been able to reproduce the error is in the axe playground page.

**Comment by WilcoFiers** [comment]
Closing, as the playground is no longer available.

@bjankord the reason for this is because these rules look up the DOM tree. If the playground is in a `<main>` element, and than a `<main>` element is put inside of that, axe-core will fail that, even if you tell axe only to test the content of the playground, that inner main will still be a dependent of the outer one.
